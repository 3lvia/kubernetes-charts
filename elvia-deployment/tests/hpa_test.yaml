suite: test hpa
templates:
  - hpa.yaml
tests:
  - it: hpa snapshot should match
    set:
      image.tag: latest
      name: demo-app
      namespace: core
      microserviceType: worker
      environment: dev
      replicaCount: 1
      resources:
        requests:
          cpu: 100m
          memory: 128Mi
        limits:
          cpu: 400m
          memory: 500Mi
      autoscaling:
        minReplicas: 2
        maxReplicas: 4
        targetCPUUtilizationPercentage: 80
    asserts:
      - matchSnapshot: {}
