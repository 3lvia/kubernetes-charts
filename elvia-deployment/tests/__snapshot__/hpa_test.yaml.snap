hpa snapshot should match:
  1: |
    apiVersion: autoscaling/v1
    kind: HorizontalPodAutoscaler
    metadata:
      name: demo-app
      namespace: core
    spec:
      maxReplicas: 4
      minReplicas: 3
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: demo-app
      targetCPUUtilizationPercentage: 80
